### PR TITLE
New resource/ data source `azurerm_storage_encryption_scope`

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/8893

=== RUN   TestAccAzureRMStorageEncryptionScope_basic
=== PAUSE TestAccAzureRMStorageEncryptionScope_basic
=== CONT  TestAccAzureRMStorageEncryptionScope_basic
--- PASS: TestAccAzureRMStorageEncryptionScope_basic (136.38s)
=== RUN   TestAccAzureRMStorageEncryptionScope_requiresImport
=== PAUSE TestAccAzureRMStorageEncryptionScope_requiresImport
=== CONT  TestAccAzureRMStorageEncryptionScope_requiresImport
--- PASS: TestAccAzureRMStorageEncryptionScope_requiresImport (146.85s)
=== RUN   TestAccAzureRMStorageEncryptionScope_completeSourceStorage
=== PAUSE TestAccAzureRMStorageEncryptionScope_completeSourceStorage
=== CONT  TestAccAzureRMStorageEncryptionScope_completeSourceStorage
--- PASS: TestAccAzureRMStorageEncryptionScope_completeSourceStorage (136.50s)
=== RUN   TestAccAzureRMStorageEncryptionScope_completeSourceKeyVault
=== PAUSE TestAccAzureRMStorageEncryptionScope_completeSourceKeyVault
=== CONT  TestAccAzureRMStorageEncryptionScope_completeSourceKeyVault
--- PASS: TestAccAzureRMStorageEncryptionScope_completeSourceKeyVault (294.01s)
=== RUN   TestAccAzureRMStorageEncryptionScope_update
=== PAUSE TestAccAzureRMStorageEncryptionScope_update
=== CONT  TestAccAzureRMStorageEncryptionScope_update
--- PASS: TestAccAzureRMStorageEncryptionScope_update (493.35s)

=== RUN   TestAccDataSourceAzureRMStorageEncryptionScope_basic
=== PAUSE TestAccDataSourceAzureRMStorageEncryptionScope_basic
=== CONT  TestAccDataSourceAzureRMStorageEncryptionScope_basic
--- PASS: TestAccDataSourceAzureRMStorageEncryptionScope_basic (140.63s)

### CODE REVIEW

**Comment by yupwei68** [comment]
I've merged the master branch. Please continue reviewing.

**Comment by tombuildsstuff** [comment]
Looks like this also requires changes to Giovanni to support the BlobEncryptionScope to be able to test this end-to-end: https://docs.microsoft.com/en-us/rest/api/storageservices/put-block

**Comment by tombuildsstuff** [comment]
For now it looks like we should be able to ship this without the Blob and Containers support, but that'll require work in both the Giovanni SDK and (subsequently) those resources to support Encryption Scopes on those resources.

File: azurerm/internal/services/storage/resource_arm_storage_encryption_scope.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+	}
+
+	props := storage.EncryptionScope{
+		EncryptionScopeProperties: &storage.EncryptionScopeProperties{
+			State: storage.Disabled,
+		},
+	}
+
+	_, err = client.Put(ctx, id.ResourceGroup, id.AccountName, id.Name, props)
+	if err != nil {
```
Comment (katbyte) [inline azurerm/internal/services/storage/resource_arm_storage_encryption_scope.go:216]:
```suggestion
	if _, err = client.Put(ctx, id.ResourceGroup, id.AccountName, id.Name, props); err != nil {
```

**Review by katbyte — APPROVED** [review]
Aside from one cOmment LGTM 👍

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-11-19 at 20 41 35

**Pull request merged**